src/template-tags/tickets.php

```diff
@@ -109,7 +109,192 @@ function tribe_events_count_available_tickets( $event = null ) {
+		);
+		$types['rsvp']    = array(
+			'count'     => 0,
+			'stock'     => 0,
+			'unlimited' => 0,
+			'available' => 0,
+		);
+
+		foreach ( $tickets as $ticket ) {
+
+			// If a ticket is not current for sale do not count it
+			if ( ! tribe_events_ticket_is_on_sale( $ticket ) ) {
+				continue;
+			}
+
+			// if ticket and not rsvp add to ticket array
+			if ( 'Tribe__Tickets__RSVP' !== $ticket->provider_class ) {
+				$types['tickets']['count'] ++;
+
+				if ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 0 === $types['tickets']['global'] ) {
+					$types['tickets']['global'] ++;
+				} elseif ( Tribe__Tickets__Global_Stock::GLOBAL_STOCK_MODE === $ticket->global_stock_mode() && 1 === $types['tickets']['global'] ) {
+					continue;
+				}
+
+				$types['tickets']['stock'] = $types['tickets']['stock'] + $ticket->stock;
+
+				if ( 0 !== $types['tickets']['stock'] ) {
+					$types['tickets']['available'] ++;
+				}
+
+				if ( ! $ticket->manage_stock() ) {
+					$types['tickets']['unlimited'] ++;
+					$types['tickets']['available'] ++;
+				}
+			} else {
+				$types['rsvp']['count'] ++;
+
+				$types['rsvp']['stock'] = $types['rsvp']['stock'] + $ticket->stock;
+				if ( 0 !== $types['rsvp']['stock'] ) {
+					$types['rsvp']['available'] ++;
+				}
+
+				if ( ! $ticket->manage_stock() ) {
+					$types['rsvp']['unlimited'] ++;
+					$types['rsvp']['available'] ++;
+				}
+			}
+		}
+
+		return $types;
+	}
+}//end if
+
+
+if ( ! function_exists( 'tribe_tickets_display_count_and_ticket_button' ) ) {
+
+	/**
+	 * Echos Remaining Ticket Count and Purchase Buttons for an Event
+	 *
+	 * @return string/null
+	 */
+	function tribe_tickets_display_count_and_ticket_button() {
+
+		$event_id = get_the_ID();
+
+		// get an array for ticket and rsvp counts
+		$types = tribe_tickets_count_tickets_and_rsvp( $event_id );
+
+		// If we have tickets or RSVP, but everything is Sold Out then display the Sold Out messages
+		if ( ( $types['tickets']['count'] || $types['rsvp']['count'] ) && ( ! $types['tickets']['available'] && ! $types['rsvp']['available'] ) ) {
+
+			$stock  = '<span class="tribe-out-of-stock">' . esc_html_x( 'Out of stock!', 'list view stock sold out', 'event-tickets' ) . '</span>';
+			$button = '<button class="tribe-button">' . esc_html_x( 'Sold Out!', 'list view sold out', 'event-tickets' ) . '</button>';
+
+			/**
+			 * Filter the ticket count and purchase button
+			 *
+			 * @var $stock    . $button stock message and button to display
+			 * @var $types    the ticket and rsvp count array for event
+			 * @var $event_id the event id
+			 */
+			echo apply_filters( 'tribe_tickets_stock_and_purchase_button', $stock . $button, $types, $event_id );
+
+			return;
+		}
+
+		$rsvp = false;
+		// Determine button text and anchor link based on an events tickets
+		if (
+				$types['tickets']['count']
+			    && $types['tickets']['available']
+			    && 0 === $types['rsvp']['count']
```

Comment:
Check out: http://moderntribe.github.io/products-engineering/php/#php-control-multi (rinse and repeat for all these if statements)